src/app/common/services/swapy-protocol.service.ts

```diff
@@ -212,31 +238,38 @@ export class SwapyProtocolService {
       });
   }
 
-  public sellAsset(contractAddress: string, value: number) {
+  public sellAssets(contractAddresses: string[], values: number[]) {
     return this.SwapyExchangeContract.methods
-      .sellAsset(contractAddress, value)
+      .sellAssets(contractAddresses, values)
       .send({
         from: this.walletService.getWallet().address,
-        gas: 150000,
+        gas: 150000 * contractAddresses.length,
         gasPrice: this.web3.utils.toWei(this.gasPrice, 'gwei')
       }).on('transactionHash', (hash) => {
         this.handleOnTransactionHash(hash);
```

Comment:
suggestion: create `handleError` to reuse this code

Reply:
Which code specifically?